component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/InputImpl.java

```diff
@@ -56,21 +72,78 @@ protected InputImpl() {
         // no-op
     }
 
+    @Override
+    public void start() {
+        super.start();
+        initCheckpointFunctions();
+    }
+
+    @Override
+    public void start(final Consumer checkpointConsumer) {
```

Comment:
why we can't add generic to the incoming consumer class?

Reply:
Consumer assumes an Object parameter passing which, I think, is fine.
Specific action, if needed, a cast maybe possible but not sure it's needed and the intent (it should only be serializating the data without knowing what's it is handling).